src/object/electrifier.cpp

```diff
@@ -17,20 +17,30 @@
 #include "object/electrifier.hpp"
 #include "supertux/sector.hpp"
 
-Electrifier::Electrifier(uint32_t oldtile, uint32_t newtile, float seconds) :
-  change_from(oldtile),
-  change_to(newtile),
+Electrifier::Electrifier(TileChangeMap replacements, float seconds) :
+  change_map(replacements),
   duration()
 {
   duration.start(seconds);
-  Sector::current()->change_solid_tiles(change_from,change_to);
+  for(auto &tile : change_map) {
+    Sector::current()->change_solid_tiles(tile.first, tile.second);
+  }
+}
+
+Electrifier::Electrifier(uint32_t oldtile, uint32_t newtile, float seconds) :
+  change_map({{oldtile, newtile}}),
+  duration()
+{
+  Electrifier(change_map, seconds);
 }
 
 void
 Electrifier::update(float )
 {
   if (duration.check()) {
-    Sector::current()->change_solid_tiles(change_to,change_from);
+    for(auto &tile : change_map){
```

Comment:
We don't have to use const here, @tobbi?

Reply:
And as I said in the general comment change it to auto& tile

Reply:
Well, I think we can use ```const``` since it doesn't change any of the contents of the array.